### PR TITLE
MCKIN-11421 Added support for python 3.5

### PR DESCRIPTION
**Pull request by nasirhjafri**

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @nasirhjafri! I've created [OSPR-3812](https://openedx.atlassian.net/browse/OSPR-3812) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by natabene** [comment]
@ihtram Please review soon.

**Comment by ihtram** [comment]
@nasirhjafri seems like you are still working and pushing changes to this PR. Please let me know once you are done with this. I will then start reviewing this.

**Comment by nasirhjafri** [comment]
@ihtram sure!

**Comment by nasirhjafri** [comment]
@ihtram please review it. Thanks!

**Comment by ihtram** [comment]
@nasirhjafri Travis is running for py27 only and we need to run it for both versions so that we can verify if unit tests are working for both py27 and py35.

File: edx_notifications/lib/publisher.py [hunk 1]
```diff
@@ -6,32 +6,32 @@
...
-from edx_notifications.renderers.renderer import (
-    register_renderer
-)
-from edx_notifications.scopes import resolve_user_scope, has_user_scope_resolver
 
-log = logging.getLogger(__name__)
+try:
+    basestring  # pylint: disable=invalid-name
+except NameError:
+    basestring = str  # pylint: disable=redefined-builtin, invalid-name
```
Comment (ihtram) [inline edx_notifications/lib/publisher.py]: Can we replace this try catch block with `basestring = six.string_types` ?

Comment (nasirhjafri) [reply edx_notifications/lib/publisher.py]: `six.string_types` is a tuple, so it doesn't work.

Comment (nasirhjafri) [reply edx_notifications/lib/publisher.py]:
```python
if six.PY3:
    basestring = str
```
This could work.

File: requirements.txt [hunk 2]
```diff
@@ -7,5 +7,5 @@ pylru==1.0.6
 git+https://github.com/edx/django-rest-framework.git@1ceda7c086fddffd1c440cc86856441bbf0bd9cb#egg=djangorestframework==3.6.3
 PyContracts==1.7.1
 underscore.py==0.1.6
-pynliner==0.5.2
+pynliner==0.8.0
```
Comment (ihtram) [inline requirements.txt]: What is reason behind this version upgrade?

File: testserver/bokchoy_settings.py [hunk 3]
```diff
@@ -168,3 +171,6 @@
 NOTIFICATION_CHANNEL_PROVIDER_TYPE_MAPS = {
     '*': 'durable',  # default global mapping
 }
+
+
+ALLOWED_HOSTS = ['*']
```
Comment (ihtram) [inline testserver/bokchoy_settings.py]: Why do we need to add allowed hosts?

File: requirements.txt (hunk 2 above)
Comment (nasirhjafri) [reply requirements.txt]: `pynliner==0.5.2` has a dependency `beautifulsoup`. `beautifulsoup3` doesn't support python 3.
`pynliner==0.8.0` used `beautifulsoup4` which supports both python3 and python2.

File: testserver/bokchoy_settings.py (hunk 3 above)
Comment (nasirhjafri) [reply testserver/bokchoy_settings.py]: I'll remove it. Not sure how it was added.

File: edx_notifications/lib/publisher.py (hunk 1 above)
Comment (ihtram) [reply edx_notifications/lib/publisher.py]:
```
if six.PY3:
    basestring = str
```
Then please use this as it seems better than try block

File: requirements.txt (hunk 2 above)
Comment (ihtram) [reply requirements.txt]: that make sense

**Comment by nasirhjafri** [comment]
@ihtram squashed the commits. I don't have merge access, Can you please merge it too?
Thanks for reviewing.

**Comment by edx-webhook** [comment]
Although this pull request is already merged, I've created [OSPR-5419](https://openedx.atlassian.net/browse/OSPR-5419) so that we can track it in Jira.

There is nothing you have to do. No action is needed from your side.  Thanks again for your contribution.

**Pull request merged**